Flakky test on scheduler

```
  io.kestra.core.schedulers.SchedulerScheduleOnDatesTest ✘ recoverNONEMissing()

    java.lang.AssertionError: 
    Expected: is <2024-09-18T21:52:13>
         but: was <2024-09-18T21:52:14>
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:6)
        at io.kestra.core.schedulers.SchedulerScheduleOnDatesTest.recoverNONEMissing(SchedulerScheduleOnDatesTest.java:254)
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        at io.micronaut.test.extensions.junit5.MicronautJunit5Extension$2.proceed(MicronautJunit5Extension.java:142)
        at io.micronaut.test.extensions.AbstractMicronautExtension.interceptEach(AbstractMicronautExtension.java:162)
        at io.micronaut.test.extensions.AbstractMicronautExtension.interceptTest(AbstractMicronautExtension.java:119)
        at io.micronaut.test.extensions.junit5.MicronautJunit5Extension.interceptTestMethod(MicronautJunit5Extension.java:129)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
```

--- a/core/src/test/java/io/kestra/core/schedulers/SchedulerScheduleOnDatesTest.java
+++ b/core/src/test/java/io/kestra/core/schedulers/SchedulerScheduleOnDatesTest.java
@@ -12,6 +12,7 @@ import io.kestra.core.utils.TestsUtils;
 import io.kestra.jdbc.runner.JdbcScheduler;
 import io.kestra.plugin.core.trigger.ScheduleOnDates;
 import jakarta.inject.Inject;
+import org.junit.jupiter.api.RepeatedTest;
 import org.junit.jupiter.api.Test;
 import reactor.core.publisher.Flux;
 
@@ -251,7 +252,9 @@ public class SchedulerScheduleOnDatesTest extends AbstractSchedulerTest {
             Await.until(() -> scheduler.isReady(), Duration.ofMillis(100), Duration.ofSeconds(5));
 
             Trigger newTrigger = this.triggerState.findLast(lastTrigger).orElseThrow();
-            assertThat(newTrigger.getNextExecutionDate().toLocalDateTime().truncatedTo(ChronoUnit.SECONDS), is(now.toLocalDateTime().truncatedTo(ChronoUnit.SECONDS)));
+            // depending on the exact timing of events, the next date can be now or after
+            assertThat(newTrigger.getNextExecutionDate().toLocalDateTime().truncatedTo(ChronoUnit.SECONDS),
+                oneOf(now.toLocalDateTime().truncatedTo(ChronoUnit.SECONDS), after.toLocalDateTime().truncatedTo(ChronoUnit.SECONDS)));
         }
     }
 }

